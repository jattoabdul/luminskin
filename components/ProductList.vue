<template>
  <div>
    <ul class="product-list">
      <li v-for="product in products" :key="product.id" class="product-list__item">
        <product :product="product" :current-currency="currentCurrency" />
      </li>
    </ul>
  </div>
</template>

<script>
import Product from './Product.vue'

export default {
  name: 'ProductList',
  components: {
    Product
  },
  props: {
    products: {
      type: Array,
      default() {
        return []
      }
    },
    currentCurrency: {
      type: String,
      default() {
        return 'USD'
      }
    }
  },
  computed: {},
  methods: {}
}
</script>

<style scoped lang="scss">
.product-list {
  padding: 0;
  display: flex;
  flex-wrap: wrap;
  justify-content: space-between;
  align-items: stretch;
}
.product-list__item {
  list-style: none;
  width: 32%;
}
</style>
