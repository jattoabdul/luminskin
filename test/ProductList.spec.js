import { shallowMount } from '@vue/test-utils'
import ProductList from '@/components/ProductList.vue'
import Vue from 'vue'
import VueCurrencyFilter from 'vue-currency-filter'
import getSymbolFromCurrency from 'currency-symbol-map'

Vue.use(VueCurrencyFilter, {
  thousandsSeparator: ',',
  fractionCount: 2,
  fractionSeparator: '.',
  symbolPosition: 'front',
  symbolSpacing: false
})

Vue.mixin({
  methods: {
    getCurrencySymbol(currency) {
      const foundCurrency = getSymbolFromCurrency(currency)
      if (foundCurrency) {
        return foundCurrency
      }
      return currency
    }
  }
})

const products = [
  {
    id: 1,
    title: 'product title',
    price: 100,
    image_url: 'image.svg'
  }
]

const factory = () => {
  return shallowMount(ProductList, {
    propsData: {
      currentCurrency: 'USD',
      products
    }
  })
}

describe('ProductList', () => {
  test('mounts properly', () => {
    const wrapper = factory()
    expect(wrapper.vm).toBeTruthy()
  })

  test('renders correctly', () => {
    const wrapper = factory()
    expect(wrapper.html()).toMatchSnapshot()
  })
})
