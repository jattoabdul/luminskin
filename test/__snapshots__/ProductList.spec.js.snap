// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductList renders correctly 1`] = `
<div>
  <ul class="product-list grid grid-cols-1 md:grid-cols-2 lg:grid-cols-3 col-gap-32 row-gap-20">
    <li class="product-list__item">
      <product-stub product="[object Object]" currentcurrency="USD"></product-stub>
    </li>
  </ul>
</div>
`;
